plugins/redis/README

```diff
@@ -0,0 +1,124 @@
+the locked key and wait for the winner to set the new value and publish the fact that it has done so.
+This is done using the redis pub/sub paradigm. Note - in this implementation all publishes contain as
+their message the very value that was just set, thus not requiring the subscriber to send subsequent GET
+(this is not just an efficiency concern).
+
+A note about the redis pub/sub mechanism is that it is completely independent from the keyspace and database
+index. Subscribers do so to a 'channel' (any given char string) only and not the particular key in question.
+Any publication on this channel is heard by all current subscribers regardless of database index. This being
+said, this plugin implementation does not allow for direct channel selection, instead it IS in fact dependent
+on the keyspace and database. This is done so in a one-to-one and unique fashion such that a given key on a
+given server:port and database creates a unique channel that is independent of plugin call and client/thread.
+
+Since redis does not implicitly support locks, it is this 'channel', itself, that represents the notion of a
+lock within this plugin library. These channels/locks are a particular string prefixed by the string
+'redis_ecl_lock'. When locked the key's value contains this channel/lock string. This is so lock acknowledging
+GETs can subscribe to the key via the correct channel, being the  locked keys vaue itself.
+
+Again, since redis does not implicitly support locks, such locking mechanisms as the above are not
+enforced/asserted by the server. They have to be honoured/acknowledged by the client instead.
+e.g. the sync module does not honour such locking and will SET and GET regardless. The async module also doesn't
+honour locks however, upon attempting to SET a locked key, it will do so but then publish to the fact that
+it has done so. This is implemented by using the redis command "GETSET key", i.e. overwriting the value regardless
+but if the overwritten value happened to be a lock it then publishes the SET value on the retrieved channel.
+
+How to use the locking module?
+
+Based on the above described scenario, ideally this would be implemented by the following function design:
+
+GET(options, key, theECLFunctionToCallUponMiss(params));
+
+Unfortunately, it is not currently possible to pass ECL functions into SERVICE definitions, such as this plugin
+is. It should also be noted that such functionality as communicating with a live and dynamic external service
+(i.e. redis) requires great precaution within the declarative and non-functional paradigm of ECL. The primary
+power to using ECL is that you tell it what you want and it figures out how to do it. These plugin function
+calls can seriously hinder this capability due to their runtime execution nature.
+
+Thus, to use the locking module two options are possible:
+
+UNSIGNED lock := locking.GetLockObject(options, key, database);
```

Comment:
Yes, this is something that we need to discuss. As I've said before this isn't actually a lock. No redis key is locked within this function. This is used to pass the required info of server:port, db, & key into the locking routines in a single 'lump' so that you can't missmash them as parameters. It then has as it's destructor a call to delete the key if still locked. I believe the plan was to return it as a row, of which kind I'm not sure.